### PR TITLE
Update aiidalab-home to 24.12.1

### PR DESCRIPTION
**Pull request by danielhollas**

### CODE REVIEW

**Comment by danielhollas** [comment]
@edan-bainglass FYI Here's how you update aiidalab-home in the AiiDAlab docker stack. 
Would you mind testing the image from this PR? (`ghcr.io/aiidalab/full-stack:pr-509`) Especially the Warning functionality. 

Are you aware of other changes in the Docker image that are needed for the Demo server? Or should I release a new version after merging this?

**Comment by edan-bainglass** [comment]
> @edan-bainglass FYI Here's how you update aiidalab-home in the AiiDAlab docker stack. Would you mind testing the image from this PR? (`ghcr.io/aiidalab/full-stack:pr-509`) Especially the Warning functionality.

Sure, as soon as you tell me how 😅 Do I create a container using `docker run ghcr.io/aiidalab/full-stack:pr-509` and test?

> Are you aware of other changes in the Docker image that are needed for the Demo server? Or should I release a new version after merging this?

Well, I am working on a new notebook to centralize the creation of new codes. Currently, we have the "Setup new code" button per code resource widget (which I made optional in a recent commit). The idea is to compensate not having it in-line by providing this centralized notebook.

Here's a snapshot of the current status:
image

@superstar54 suggested correctly that this **should not** be tied to any one app, allowing the user to create **any** code. With that in mind, I was considering then to open the PR in the aiidalab-home repo. So...

1. Thoughts?
2. If agreed, on merge, will this force another release? If so, I'm quite close to a stable product.

Let me know 🙂

**Comment by danielhollas** [comment]
> docker run ghcr.io/aiidalab/full-stack:pr-509

Yes, this should work, though I think you'd also need to expose the jupyter port, something like 

```console
docker run -p 8888:8888 ghcr.io/aiidalab/full-stack:pr-509
```

But in general I highly recommend using the `aiidalab-launch` app that was created to abstract away the direct docker invocation. Please have a look at the README.

https://github.com/aiidalab/aiidalab-launch?tab=readme-ov-file#getting-started

> Well, I am working on a new notebook to centralize the creation of new codes. Currently, we have the "Setup new code" button per code resource widget (which I made optional in a recent commit). The idea is to compensate not having it in-line by providing this centralized notebook.

This feels like it would go either to QeApp or AWB, neither of which are currently pre-installed in the base image so it's not a blocker here.

> With that in mind, I was considering then to open the PR in the aiidalab-home repo. So...

Hmm, but where would you put it? We currently don't have a place for it in aiidalab-home. There's an open PR on aiidalab-home for a Control page where I think this should go, but I think that PR should be landed first.

https://github.com/aiidalab/aiidalab-home/pull/156

**Comment by edan-bainglass** [comment]
@danielhollas sorry, should've come back to see your comment prior to opening the PR. Anyhow, will think more about a proper placement.

**Comment by edan-bainglass** [comment]
> @edan-bainglass FYI Here's how you update aiidalab-home in the AiiDAlab docker stack. Would you mind testing the image from this PR? (`ghcr.io/aiidalab/full-stack:pr-509`) Especially the Warning functionality.

I'm confused. The warning functionality is not implemented in this release.

**Comment by danielhollas** [comment]
Huh, that's weird. Could you please investigate a bit more? The code for the home app inside the container is in `/opt/home` I believe (accessible via `~/apps/home` symlink). You can check if the code for the warning is there. I can't see how it could be missing there, I just double checked that I made the release of `aiidalab-home` properly (it's just a tag on a commit).

**Comment by edan-bainglass** [comment]
> Huh, that's weird. Could you please investigate a bit more? The code for the home app inside the container is in `/opt/home` I believe (accessible via `~/apps/home` symlink). You can check if the code for the warning is there. I can't see how it could be missing there, I just double checked that I made the release of `aiidalab-home` properly (it's just a tag on a commit).

Probably not good that `git branch` yields `HEAD detached at v24.09.0` 🤔 Should be `v24.12.0`, no?

**Comment by danielhollas** [comment]
> Should be v24.12.0, no?
Indeed it should. Are you sure you're running the correct image (`ghcr.io/aiidalab/full-stack:pr-509`).
Is the home app linked to `/opt/home`?

**Comment by edan-bainglass** [comment]
> > Should be v24.12.0, no?
> 
> Indeed it should. Are you sure you're running the correct image (`ghcr.io/aiidalab/full-stack:pr-509`). Is the home app linked to `/opt/home`?

Oops, was pulling `ghcr.io/aiidalab/full-stack`, not `:pr-509`. My bad 🙏

**Comment by edan-bainglass** [comment]
@danielhollas there appears to be an issue with my warning message. When I add the `.aiidalab/home_app_warning.md` file, instead of adding the message after the home's control panel, it replaces the panel.

Before adding file
image

After adding file
image

I think I found the issue. While working on the in-app guides PR, I found a more reliable approach to loading local files using the `markdown` module instead of `jinja2`. I tested it quickly here on the pr-509 container and it works.

image

I'll PR a fix. Apologies for having to create a new release for this. Maybe we can do it together this time?

**Comment by danielhollas** [comment]
> Apologies for having to create a new release for this. Maybe we can do it together this time?

No problem, feel free to go ahead with the release, the instructions are here:

https://github.com/aiidalab/aiidalab-home?tab=readme-ov-file#for-maintainers

tl;dr basically for aiidalab-home making a new release is just creating a new tagged commit and bumping version in setup.cfg, and we use the `bumpver` tool to automate that, as described in the instruction (always run with `--dry` option first to inspect the changes.). Please coordinate with Jusong if anything is unclear.

One big gotcha is that one has to push directly to main branch. I believe you now have the necessary permissions for that, but you might want to test that by pushing some trivial commit first (e.g. modifying / improving the README file).

**Comment by edan-bainglass** [comment]
> No problem, feel free to go ahead with the release, the instructions are here:
> 
> https://github.com/aiidalab/aiidalab-home?tab=readme-ov-file#for-maintainers
> 
> tl;dr basically for aiidalab-home making a new release is just creating a new tagged commit and bumping version in setup.cfg, and we use the `bumpver` tool to automate that, as described in the instruction (always run with `--dry` option first to inspect the changes.). Please coordinate with Jusong if anything is unclear.
> 
> One big gotcha is that one has to push directly to main branch. I believe you now have the necessary permissions for that, but you might want to test that by pushing some trivial commit first (e.g. modifying / improving the README file).

Got it! Thanks @danielhollas. I'll coordinate with @unkcpz if necessary. Happy holidays 🙂

File: build.json [hunk 1]
```diff
@@ -16,7 +16,7 @@
       "default": "24.09.0"
     },
     "AIIDALAB_HOME_VERSION": {
-      "default": "24.09.0"
+      "default": "24.12.0"
```
Comment (edan-bainglass) [inline build.json:19]:
```suggestion
      "default": "24.12.1"
```
